nibabel/streamlines/tractogram.py

```diff
@@ -136,6 +162,32 @@ def __setitem__(self, key, value):
 
         self.store[key] = value
 
+    def extend(self, other):
```

Comment:
I had to duplicate it (almost every line except one) because one method is using `self[key] = np.concatenate([self[key], other[key]])` and the other `self[key].extend(other[key])`. If you think of an alternative, I'm all ears.